run.go

```diff
@@ -452,31 +462,559 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 	if err != nil {
 		return errors.Wrapf(err, "error resolving mountpoints for container")
 	}
-	specbytes, err := json.Marshal(spec)
+	return b.runUsingRuntimeSubproc(options, spec, mountPoint, path, Package+"-"+filepath.Base(path))
+}
+
+type runUsingRuntimeSubprocOptions struct {
+	Options       RunOptions
+	Spec          *specs.Spec
+	RootPath      string
+	BundlePath    string
+	ContainerName string
+}
+
+func (b *Builder) runUsingRuntimeSubproc(options RunOptions, spec *specs.Spec, rootPath, bundlePath, containerName string) (err error) {
+	var confwg sync.WaitGroup
+	config, conferr := json.Marshal(runUsingRuntimeSubprocOptions{
+		Options:       options,
+		Spec:          spec,
+		RootPath:      rootPath,
+		BundlePath:    bundlePath,
+		ContainerName: containerName,
+	})
+	if conferr != nil {
+		return errors.Wrapf(conferr, "error encoding configuration for %q", runUsingRuntimeCommand)
+	}
+	cmd := reexec.Command(runUsingRuntimeCommand)
```

Comment:
@nalind Why are you using `reexec` here, instead of continuing on with the code in the same process? The code itself looks fine, just curious as to why you're doing this in a fresh main.

Reply:
Nevermind, I see a comment below about setting subreaper - makes sense now.